ingestion-edge/tests/test_api.py

```diff
@@ -4,107 +4,59 @@
-            subscriber.create_subscription(subscription, topic)
-
-    # test route table
-    for path, topic, methods in ROUTE_TABLE:
-        subscription = topic_to_subscription(topic)
-        for method in methods:
-            # fast forward subscriber to HEAD
-            while True:
-                resp = subscriber.pull(
-                    subscription,
-                    100,
-                    timeout=0.1,
-                )
-                if not resp.received_messages:
-                    break
-                subscriber.acknowledge(
-                    subscription,
-                    [message.ack_id for message in resp.received_messages],
-                )
-
-            # submit request to edge
-            uri = path.replace("<path:suffix>", "test")
-            body = "test"
-            req_time = datetime.utcnow()
-            r = getattr(requests_session, method.lower())(server + uri, body)
-            r.raise_for_status()
-
-            # validate message delivered to pubsub
-            resp = subscriber.pull(subscription, 1, timeout=0.1)
-            assert len(resp.received_messages) == 1
-            assert resp.received_messages[0].message.data == b"test"
-            attributes = resp.received_messages[0].message.attributes
-            expect = {
-                "args": "",
-                "content_length": str(len(body)),
-                "host": server.split("://", 1)[1],
-                "method": method,
-                "protocol": server.split("://", 1)[0],
-                "remote_addr": lambda x: True,
-                "submission_timestamp": lambda x: (
-                    req_time - parse(x[:-1])
-                ).total_seconds() < 1,
-                "uri": uri,
-                "user_agent": "python-requests/" + requests.__version__,
-            }
-            for key, validate in expect.items():
-                assert key in attributes
-                if type(validate) is str:
-                    assert attributes[key] == validate
-                else:
-                    assert validate(attributes[key])
+from flask.testing import FlaskClient
+from typing import Dict, List, Tuple
+from unittest.mock import patch
+
+
+def test_dockerflow(client: FlaskClient):
+    r = client.get("/__heartbeat__")
+    assert r.status_code == 200
+    assert r.json == {"checks": {}, "details": {}, "status": "ok"}
+
+    r = client.get("/__lbheartbeat__")
+    assert r.status_code == 200
+    assert not r.data
+
+    r = client.get("/__version__")
+    assert r.status_code == 200
+    assert sorted(r.json.keys()) == ["build", "commit", "source", "version"]
+
+
+def test_publish(client: FlaskClient):
+    from ingestion_edge.config import ROUTE_TABLE
+
+    messages: List[Tuple[str, bytes, Dict[str, str]]] = []
+
+    def _publish(*args):
+        messages.append(args)
+
+    with patch('ingestion_edge.publish._publish', new=_publish):
+        # test route table
+        for route in ROUTE_TABLE:
```

Comment:
Would it make sense to use [`@pytest.mark.parametrize`](https://docs.pytest.org/en/latest/parametrize.html) to provides the route instead of looping over `ROUTE_TABLE`? Doing that would make each route its own test, and they could fail independently. I'm not sure what you're testing here, so maybe that's not useful.